Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h

```diff
@@ -21,19 +21,47 @@ class PROGRESSIONSYSTEMRUNTIME_API APSStarActor : public AActor
 	/** Function called every frame on this Actor */
 	virtual void Tick(float DeltaTime) override;
 
-	/** Stores the starting time to hide stars */
+	/** Stores the starting time to hide stars in the main menu when cinematic started */
 	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Starting time to hide stars"))
 	float StartTimeHideStarsInternal = 0.0f;
 
+	/** Stores the starting time to animate stars in main menu */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Starting time to animate stars in menu"))
+	float StartTimeMenuStarsInternal = 0.0f;
+
 	/** When a local character load finished */
 	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
 	void OnLocalCharacterReady(APlayerCharacter* Character, int32 CharacterID);
 
-	/** Is called when any cinematic started */
+	/** Called when the current player was changed */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void OnPlayerTypeChanged(FPlayerTag PlayerTag);
+
+	/** Called when the current game state was changed */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void OnGameStateChanged(ECurrentGameState GameState);
+
+	/** Is called when any cinematic started to play in the main menu */
 	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
 	void OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator);
 
-	/** Hiding stars with animation */
+	/** Is called to set a start time for animations */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void SetAnimationStartTime(float& StartTime);
+
+	/** Called to initialize the Star animation in the main menu */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void InitStarMenuAnimation();
+
+	/** Hiding stars with animation in main menu when cinematic is start to play */
 	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
 	void TryPlayHideStarAnimation();
+
+	/** Menu stars with animation in main menu idle */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void TryPlayMenuStarAnimation();
+
+	/** Playing star animation */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void TryPlayStarAnimation(float& StartTime, UCurveTable* AnimationCurveTable);
```

Comment:
1. please make it `static` as it's helper without any dependency to the class itself
2. All non-const reference parameters have to be named as Out-prefix or Ref-postfix: please rename your `StartTime` to be `StartTimeRef`
2. Do `UPARAM(ref) float& StartTimeRef` instead of `float& StartTimeRef`, otherwise blueprints will show this node with StartTime as return parameter.

Expected result:
```
static void TryPlayStarAnimation(UPARAM(ref) float& StartTimeRef, UCurveTable* AnimationCurveTable);
```

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/6141759e381d2b8658971efac878aaebce1a9cf7